src/pages/SignIn/SignIn.jsx

```diff
@@ -1,19 +1,44 @@
 import "./SignIn.css";
+import { useState, useCallback } from "react";
 import { useNavigate } from "react-router-dom";
+import Input from "../../components/Input/Input";
 
 const SignIn = () => {
   const navigate = useNavigate();
+  const [isFormValid, setIsFormValid] = useState({
+    username: false,
+    password: false,
+  });
 
-  const goHomeHandler = () => navigate("/home");
+  const goHomeHandler = (event) => {
+    event.preventDefault();
+    navigate("/home");
+  };
+
+  const validateFormHandler = useCallback((isValid, id) => {
+    setIsFormValid((prevState) => {
+      const prevErrors = { ...prevState };
+      prevErrors[id] = isValid;
+      return prevErrors;
+    });
+  }, []);
+
+  const isButtonDisabled = Object.values(isFormValid).includes(false);
 
   return (
     <div className="sign-in-background">
       <form onSubmit={goHomeHandler} className="sign-in-form">
-        <label htmlFor="username">Username</label>
-        <input id="username" type="text" className="input" />
-        <label htmlFor="password">Password</label>
-        <input id="password" type="password" className="input" />
-        <button type="submit" className="sign-in-button">
+        <Input id="username" type="text" validateForm={validateFormHandler}>
+          Username
+        </Input>
+        <Input id="password" type="password" validateForm={validateFormHandler}>
+          Password
+        </Input>
```

Comment:
Very static. I would suggest to create a configuration object/array of objects with provided fields' details and then use map/forEach to render them properly.